# Title

## Introduction

*some description about the data and the projects*

In [24]:
# Import Packages
import pandas as pd
import numpy as np

## Data Cleaning/Loading

### Dataset Loading

In [25]:
# Load Dataset
objects = r"C:\Users\Yusuf Saputra\OneDrive - University of Toronto\Documents\UofT\Winter 2024\ECO225\ECO225 Project\Data\startup\objects.csv"
df_objects = pd.read_csv(objects)

C:\Users\Yusuf Saputra\AppData\Local\Temp\ipykernel_8724\2591217029.py:3: DtypeWarning: Columns (3,7,9,10,17,18,21,22,23,25,26,29,30,37) have mixed types. Specify dtype option on import or set low_memory=False.
  df_objects = pd.read_csv(objects)


In [27]:
# Quick check on the composition of the dataset

df_objects.shape

(462651, 40)

This tells us that the dataset that we choose contains 462,651 rows x 40 columns.

In [28]:
# Taking snapshot of the dataset
df_objects.head()

id entity_type  entity_id parent_id                name  \
0      c:1     Company          1       NaN            Wetpaint   
1     c:10     Company         10       NaN             Flektor   
2    c:100     Company        100       NaN               There   
3  c:10000     Company      10000       NaN             MYWEBBO   
4  c:10001     Company      10001       NaN  THE Movie Streamer   

      normalized_name                    permalink    category_code  \
0            wetpaint            /company/wetpaint              web   
1             flektor             /company/flektor      games_video   
2               there               /company/there      games_video   
3             mywebbo             /company/mywebbo  network_hosting   
4  the movie streamer  /company/the-movie-streamer      games_video   

      status  founded_at  ... last_funding_at funding_rounds  \
0  operating  2005-10-17  ...      2008-05-19              3   
1   acquired         NaN  ...             NaN              0   
2   acquired         NaN  ...             NaN              0   
3  operating  2008-07-26  ...             NaN              0   
4  operating  2008-07-26  ...             NaN              0   

  funding_total_usd first_milestone_at last_milestone_at  milestones  \
0        39750000.0         2010-09-05        2013-09-18           5   
1               0.0                NaN               NaN           0   
2               0.0         2003-02-01        2011-09-23           4   
3               0.0                NaN               NaN           0   
4               0.0                NaN               NaN           0   

   relationships        created_by           created_at           updated_at  
0             17  initial-importer  2007-05-25 06:51:27  2013-04-13 03:29:00  
1              6  initial-importer  2007-05-31 21:11:51  2008-05-23 23:23:14  
2             12  initial-importer  2007-08-06 23:52:45  2013-11-04 02:09:48  
3              0               NaN  2008-08-24 16:51:57  2008-09-06 14:19:18  
4              0               NaN  2008-08-24 17:10:34  2008-09-06 14:19:18  

[5 rows x 40 columns]

*some description about the table (i.e., Missing Values, Types of Values)*

### Data Cleaning

*some explanation on how we clean (i.e., because of the missing value)*

In [29]:
# Find Missing Value
df_objects.isna().sum()

id                          0
entity_type                 0
entity_id                   0
parent_id              434936
name                        4
normalized_name            31
permalink                   0
category_code          339465
status                      0
founded_at             362210
closed_at              459842
domain                 287709
homepage_url           287709
twitter_username       336562
logo_url               253801
logo_width                  0
logo_height                 0
short_description      455034
description            367646
overview               227416
tag_list               356155
country_code           367608
state_code             407891
city                   371967
region                      0
first_investment_at    445695
last_investment_at     445695
investment_rounds           0
invested_companies          0
first_funding_at       431144
last_funding_at        431144
funding_rounds              0
funding_total_usd           0
first_mile

*description*

In [34]:
# Drop the rows that have missing values in some particular columns
df_clean = df_objects[~(
    df_objects['category_code'].isnull() |
    df_objects['country_code'].isnull() | 
    df_objects['founded_at'].isnull() |
    df_objects['closed_at'].isnull()
    
)]

# Check
df_clean.isna().sum()

id                        0
entity_type               0
entity_id                 0
parent_id              1661
name                      0
normalized_name           0
permalink                 0
category_code             0
status                    0
founded_at                0
closed_at                 0
domain                   18
homepage_url             18
twitter_username        831
logo_url                129
logo_width                0
logo_height               0
short_description      1206
description             453
overview                 15
tag_list                568
country_code              0
state_code              605
city                     78
region                    0
first_investment_at    1655
last_investment_at     1655
investment_rounds         0
invested_companies        0
first_funding_at        312
last_funding_at         312
funding_rounds            0
funding_total_usd         0
first_milestone_at      702
last_milestone_at       702
milestones          

*description*

### Summary Statistics

In [35]:
df_objects['funding_total_usd'].describe()

count    4.626510e+05
mean     8.926722e+05
std      1.700120e+07
min      0.000000e+00
25%      0.000000e+00
50%      0.000000e+00
75%      0.000000e+00
max      5.700000e+09
Name: funding_total_usd, dtype: float64

*description*